Assets/Scripts/Mixed/Systems/PlayerPhysicsBasedMovement.cs

```diff
@@ -100,21 +306,49 @@ protected override void OnUpdate()
                 // Make movement vector based on player input
                 float3 movementVelocity = math.mul(horizPlaneView, direction) * speedMultiplier;
 
-                // Preserve current vertical movement
-                movementVelocity.y = pv.Linear.y;
-
                 // Check if is grounded for jumping
-                bool grounded = ISGrounded(trans, settings.maxWalkAngle, settings.groundCheckDistance);
+                float2 groundedCheck = this.AngleBetweenGround(trans.Value, settings.groundCheckDistance, collider, ent.Index, rot.Value);
+                float dist = groundedCheck.x;
+                float angle = groundedCheck.y;
+                bool grounded = dist >= 0 && angle < settings.maxWalkAngle;
```

Comment:
Will the distance  from the ground to the player always be represented as a negative value? Or why are you also taking positive values to represent the player is grounded?

Reply:
This will return a -1 if the ground distance is more than the ground check distance. I will adjust this to use a struct with thee parameters, distance to ground, is within check distance, and angle of hit

Reply:
Player can jump if they are within ground check distance of the ground. It's weird and will make more sense when I break this into two systems of KCC Gravity and KCC Jump